src/api/FactoryAPI.ts

```diff
@@ -105,6 +110,41 @@ export class FactoryAPI {
     return extractNewSetTokenAddressFromLogs(transactionLogs);
   }
 
+  /**
+   * Calculates the minimum allowable natural unit for a list of ERC20 component addresses
+   * where the minimum natural unit allowed equal 10^(18-minDecimal)
+   *
+   * @param componentAddresses    Component ERC20 addresses
+   *
+   * @return                      The minimum value of the natural unit allowed by component decimals
+   */
+  public async calculateMinimumNaturalUnit(
+    components: Address[],
+  ): Promise<BigNumber> {
+    const componentInstancePromises = _.map(components, component => {
```

Comment:
Instead of instantiating the instance and then getting the decimals, why not import ERC20Wrapper and call the decimal function directly?